### DISCUSSION TITLE
How to make the app.js smaller?

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [vuepress/core#1128] [Q&A] [answered]
Hello,

I see this warning during the Vite build:

```sh
⠇ Compiling with vite
(!) Some chunks are larger than 500 KiB after minification. Consider:        
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/guide/en/#outputmanualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
```

I haven't found a discussion on this, still I have looked at a few resources, none providing a solution because my large file isn't from the vendors code.

When I look at the website, I see that the app file is the file > 500kb:

![image](https://user-images.githubusercontent.com/2410556/196021101-7b663cce-8395-427b-8d4b-cf1ea2f98aa6.png)

When I look into the file, I see quite a bit of content.

I'd like to know and understand how @Mister-Hope achieve [his small app file](https://mister-hope.github.io/assets/js/runtime~app.0d0389b0.js) with the reference to individual js files for the content (at least that is what I think I understand from looking at it).

![image](https://user-images.githubusercontent.com/2410556/196021263-b284eefd-b2a3-4f9e-a0b4-30d902be59eb.png)

Thanks a million!

My website: https://iamjeremie.me/
My repo: https://github.com/JeremieLitzler/journal

**Comment by azat-io** [comment, marked as answer]
Currently you are generating temp file with data of all your pages included your content:

```js
    const pagesData = app.pages.map((page) => {
      console.log(page.filePathRelative);
      return page;
    });
    await app.writeTemp(
      'pages.js',
      `export default [${JSON.stringify(pagesData)}]`,
    );
```

And after then you are importing this file in your client code in `components/PostsIndex.vue`. This is the reason why your bundle is huge.

I tried to exclude data from the bundle that is not used in this component:
https://github.com/JeremieLitzler/journal/pull/31

Thus, I managed to reduce the size from 6.9M to 1.2M. 🎉 

But I still think that this solution is not the best solution.

**Reply by JeremieLitzler** [reply]
thanks @azat-io. I'm satisfied with this solution for now.
